[GTK][WPE] Add WebKitWebFormManager and deprecate WebKitWebPage form related signals and WebKitDOMElement

carlosgcampos opened the pull request:
#### 81f661ea4d757aa254c322965cba2c1e7881491f
<pre>
[GTK][WPE] Add WebKitWebFormManager and deprecate WebKitWebPage form related signals and WebKitDOMElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=248657">https://bugs.webkit.org/show_bug.cgi?id=248657</a>

Reviewed by Michael Catanzaro.

Move the form signals from WebKitWebPage to WebKitWebFormManager that no longer uses WebKitDOMElement, but JSCValue.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMElement.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(webkitFrameGetJSCValueForElementInWorld):
(webkitFrameGetJSCValuesForElementsInWorld):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFramePrivate.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp: Added.
(webkit_web_form_manager_class_init):
(webkitWebFormManagerCreate):
(webkitWebFormManagerDidAssociateFormControls):
(webkitWebFormManagerWillSendSubmitEvent):
(webkitWebFormManagerWillSubmitForm):
(nodeForJSCValue):
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.h.in: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManagerPrivate.h: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(worldDestroyed):
(webkitWebPageDispose):
(webkit_web_page_class_init):
(webkit_web_page_get_form_manager):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMElement.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/DOMElementTest.cpp:
(DOMElementTest::testAutoFill):
* Tools/TestWebKitAPI/Tests/WebKitGLib/FrameTest.cpp:
(WebKitFrameTest::testJavaScriptValues):
(WebKitFrameTest::willSubmitFormCallback):
(WebKitFrameTest::testSubframe):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp:
(didAssociateFormControlsCallback):
(testWebExtensionFormControlsAssociated):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(formControlsAssociatedForFrameCallback):
(formControlsAssociatedCallback):
(willSubmitFormDeprecatedCallback):
(handleFormSubmissionCallback):
(willSendSubmitEventCallback):
(willSubmitFormCallback):
(pageCreatedCallback):
(windowObjectCleared):
(methodCallCallback):
(webkit_web_extension_initialize_with_user_data):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/257366@main">https://commits.webkit.org/257366@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f1cfbbf57e8c0b0d03c59d9d2937746b995a478

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98750 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7961 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31883 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108166 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168420 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102702 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85329 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91278 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106083 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104432 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6433 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89991 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33414 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21334 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1878 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22863 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1785 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45340 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6734 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42318 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2544 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3180 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
```diff
@@ -0,0 +1,248 @@
+ *
+ * Get whether @element is an HTML text input element that has been edited by a user action.
+ *
+ * Returns: %TRUE if @element is an HTML text input element tha has been edited by a user action,
```
```suggestion
 * Returns: %TRUE if @element is an HTML text input element that has been edited by a user action,
```

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
```diff
@@ -0,0 +1,248 @@
+ * @element: a #JSCValue
+ * @value: the text to set
+ *
+ * Set the value of an HTML input element as if it had been edited by
+ * the user, triggering a change event, and set it as filled automatically.
+ * If @element is not an HTML input element this function does nothing.
+ *
+ * Since: 2.40
+ */
+void webkit_web_form_manager_input_element_auto_fill(JSCValue* element, const char* value)
```
I'll be eager to test if this actually works. Currently I have to type and delete a character to get many websites to accept my autofilled usernames/passwords.

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp:
```diff
@@ -0,0 +1,248 @@
+
+    auto* node = nodeForJSCValue(element);
+    if (!is<WebCore::HTMLInputElement>(node))
+        return;
```
I think a g_warning() would be appropriate here, so it doesn't fail silently. This will make problems way easier to debug.

carlosgcampos committed:
[GTK][WPE] Add WebKitWebFormManager and deprecate WebKitWebPage form related signals and WebKitDOMElement
https://bugs.webkit.org/show_bug.cgi?id=248657

Reviewed by Michael Catanzaro.

Move the form signals from WebKitWebPage to WebKitWebFormManager that no longer uses WebKitDOMElement, but JSCValue.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMElement.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(webkitFrameGetJSCValueForElementInWorld):
(webkitFrameGetJSCValuesForElementsInWorld):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFramePrivate.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.cpp: Added.
(webkit_web_form_manager_class_init):
(webkitWebFormManagerCreate):
(webkitWebFormManagerDidAssociateFormControls):
(webkitWebFormManagerWillSendSubmitEvent):
(webkitWebFormManagerWillSubmitForm):
(nodeForJSCValue):
(webkit_web_form_manager_input_element_is_user_edited):
(webkit_web_form_manager_input_element_auto_fill):
(webkit_web_form_manager_input_element_is_auto_filled):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManager.h.in: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebFormManagerPrivate.h: Added.
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(worldDestroyed):
(webkitWebPageDispose):
(webkit_web_page_class_init):
(webkit_web_page_get_form_manager):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-extension.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMElement.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/DOMElementTest.cpp:
(DOMElementTest::testAutoFill):
* Tools/TestWebKitAPI/Tests/WebKitGLib/FrameTest.cpp:
(WebKitFrameTest::testJavaScriptValues):
(WebKitFrameTest::willSubmitFormCallback):
(WebKitFrameTest::testSubframe):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebExtensions.cpp:
(didAssociateFormControlsCallback):
(testWebExtensionFormControlsAssociated):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(formControlsAssociatedForFrameCallback):
(formControlsAssociatedCallback):
(willSubmitFormDeprecatedCallback):
(handleFormSubmissionCallback):
(willSendSubmitEventCallback):
(willSubmitFormCallback):
(pageCreatedCallback):
(windowObjectCleared):
(methodCallCallback):
(webkit_web_extension_initialize_with_user_data):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/257366@main
